packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -436,6 +448,43 @@ function isConditionalTest(node: TSESTree.Node): boolean {
   return false;
 }
 
+function isMakeBoolean(node: TSESTree.Node): boolean {
+  const parents = new Set<TSESTree.Node | null>([node]);
+  let current = node.parent;
+  while (current) {
+    parents.add(current);
+
+    if (
+      (current.type === AST_NODE_TYPES.UnaryExpression &&
+        current.operator === '!') ||
+      (current.type === AST_NODE_TYPES.CallExpression &&
+        current.callee.type === AST_NODE_TYPES.Identifier &&
+        // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+        current.callee.name === 'Boolean')
```

Comment:
[Bug] Folks sometimes write their own `function Boolean( ... ) { ... }`. This rule will need to handle that well.

Previous example: https://github.com/typescript-eslint/typescript-eslint/pull/10005#discussion_r1771754649